frontend/__tests__/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/SingleSelection.test.tsx

```diff
@@ -4,30 +4,53 @@ import userEvent from '@testing-library/user-event'
 import { Provider } from 'react-redux'
 import { store } from '@src/store'
 
+const mockValidationCheckContext = {
+  errorMessages: [],
+  clearErrorMessage: jest.fn(),
+  checkDuplicatedPipeLine: jest.fn(),
+  checkPipelineValidation: jest.fn(),
+}
+
+jest.mock('@src/hooks/useMetricsStepValidationCheckContext', () => ({
+  useMetricsStepValidationCheckContext: () => mockValidationCheckContext,
+}))
+
 describe('SingleSelection', () => {
   const mockOptions = ['mockOptions 1', 'mockOptions 2', 'mockOptions 3']
```

Comment:
Sometimes the setup is written outside the description, and sometimes it is written inside. Perhaps it should be unified